### PR TITLE
Http acceptor multi factor acceptor tests

### PR DESCRIPTION
**Pull request by dpwspoon**
- Added acceptor tests
- Refactored HttpRealmInfo to remove authorization mode

### CODE REVIEW

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java [hunk 1]
```diff
@@ -373,6 +374,18 @@ protected boolean login(NextFilter nextFilter,
                     }
                     // Login was successful, but the subject identified does not have required roles
                     // to create the web socket.  We re-issue a challenge here.
```
Comment (jfallows) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java]: This comment needs to move below codeblock, right?

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java [hunk 2]
```diff
@@ -373,6 +374,18 @@ protected boolean login(NextFilter nextFilter,
...
+                        loginContexts[realmIndex] = loginContext;
+
+                        // remember login context
+                        httpRequest.setLoginContext(loginContext);
+
+                        // remember subject
+                        httpRequest.setSubject((loginContext == null || loginContext == LOGIN_CONTEXT_OK) ? subject : loginContext.getSubject());
+//                        realmIndex++;
+                        return true;
+                    }
```
Comment (jfallows) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java]: Does `resultType` `SUCCESS` mean that all factors have succeeded or just the current factor?

If it means only the current factor has succeeded, then this seems to be returning `true` eagerly, even before all factors have been processed, right?

File: resource.address/http/src/main/java/org/kaazing/gateway/resource/address/http/HttpRealmInfo.java [hunk 3]
```diff
@@ -24,7 +24,6 @@
 public class HttpRealmInfo {
```
Comment (jfallows) [inline resource.address/http/src/main/java/org/kaazing/gateway/resource/address/http/HttpRealmInfo.java]: Recommend splitting this into a `HttpRealmInfo` interface with getters only and a `DefaultHttpRealmInfo` implementation with setters and getters in an internal package, then mock interface from tests.

File: transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java (hunk 1 above)
Comment (dpwspoon) [inline transport/http/src/main/java/org/kaazing/gateway/transport/http/bridge/filter/HttpLoginSecurityFilter.java]: fixed

**Pull request merged**